third_party/maya/lib/usdMaya/usdImport.cpp

```diff
@@ -109,18 +111,28 @@ MStatus usdImport::doIt(const MArgList & args)
         MString tmpVal;
         argData.getFlagArgument("file", 0, tmpVal);
 
-        // resolve the path into an absolute path
-        MFileObject absoluteFile;
-        absoluteFile.setRawFullName(tmpVal);
-        absoluteFile.setRawFullName( absoluteFile.resolvedFullName() ); // Make sure an absolute path
-
-        if (!absoluteFile.exists()) {
-            MGlobal::displayError("File does not exist.  Exiting.");
+        // First resolve using maya's resolver, to expand any environment
+        // variables, or any maya-specific tokens
+        MFileObject mayaFile;
+        mayaFile.setRawFullName(tmpVal);
+
+        // Use the usd resolver - this should also have the side effect of making
+        // absolute paths
+        std::string expandedPath = mayaFile.expandedFullName().asChar();
+        ArResolver& resolver = ArGetResolver();
+        resolver.ConfigureResolverForAsset(expandedPath);
```

Comment:
As far as configuring the resolver, that wasn't done in response to a specific problem, but rather just confusion about best practices - again, see my first question here: https://groups.google.com/forum/#!topic/usd-interest/E3huSatYrLI

Your comment about this being a stateful change helps clarify things somewhat, though - on re-reviewing the existing code, it seems like everywhere that's calling ConfigureResolverForAsset is doing so from within "application level" code.